sql/sql_acl.cc

```diff
@@ -14313,24 +14313,26 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
-  else if (redirect_enabled)
+  else
   {
-    char msg[MAX_REDIRECTION_LEN];
-    size_t len = 0;
+    char *msg = NULL;
+
+    if (redirect_enabled)
+    {
+      msg = new char[MAX_REDIRECTION_LEN];
```

Comment:
there is a "new" but there is no "delete",  which looks like a memory leak. I think what you could do, for example to avoid the leak have a msg_buf [MAX_REDIRECTION_LEN], and msg pointer, which points  NULL normally, but could also point to msg_buf (which would contain formatted message, if redirection is enabled, and the size of the message is OK)

Reply:
Thanks for the suggestion! Have made the change accordingly.